### PR TITLE
Move version.txt inside spinalcordtoolbox package

### LINKED ISSUE #2832: Fetching version.txt from site-package does not work for RTD
Context: https://github.com/neuropoly/spinalcordtoolbox/pull/2814#discussion_r466113927

In `spinalcordtoolbox.utils`, the version is fetched using `os.path.join(__sct_dir__, 'version.txt')`, which, for a pip-installed package, is pointing to `envs/sct-env/lib/python3.6/site-packages/version.txt`, but the file does not exist there.

A solution (anticipating a `pip install spinalcordtoolbox` in the long term) would be to:
- move `version.txt` inside `spinalcordtoolbox/`, so it is automatically copied with pip install
- edit `spinalcordtoolbox.utils` to point to this new location

### PR DESCRIPTION
**Pull request by jcohenadad**
Context: https://github.com/neuropoly/spinalcordtoolbox/pull/2814#discussion_r466113927

In `spinalcordtoolbox.utils`, the version is fetched using `os.path.join(__sct_dir__, 'version.txt')`, which, for a pip-installed package, is pointing to `envs/sct-env/lib/python3.6/site-packages/version.txt`, but the file does not exist there.

This causes the RTD script function to not show the usage:

image

The solution implemented in this PR is to:
- move `version.txt` inside `spinalcordtoolbox/`, so it is automatically copied with pip install
- edit `spinalcordtoolbox.utils` to point to this new location

Partly addresses https://github.com/neuropoly/spinalcordtoolbox/issues/1526#issuecomment-411509189
Fixes #2832

### CODE REVIEW

**Comment by jcohenadad** [comment]
@joshuacwnewton i guess we can close this PR as https://github.com/neuropoly/spinalcordtoolbox/pull/2837 does include my changes-- right?

**Comment by joshuacwnewton** [comment]
@jcohenadad I was planning on updating #2837 when this one gets merged, just to keep things organized.

**Comment by jcohenadad** [comment]
@kousu travis-windows fails: https://travis-ci.org/github/neuropoly/spinalcordtoolbox/jobs/716982434

do you think it is related to changes in this PR?

**Comment by joshuacwnewton** [comment]
@jcohenadad I don't believe so.

It looks like there was a change on Microsoft's end. The install script Chocolatey package (`wsl-ubuntu-1804`) fetches WSL as an `.appx` file from Microsoft at this link: https://aka.ms/wsl-ubuntu-1804 (More information [here](https://docs.microsoft.com/en-us/windows/wsl/install-manual).)

On the previous successful test, the downloaded `.appx` file was `CanonicalGroupLimited.Ubuntu18.04onWindows_1804.2018.817.0_x64__79rhkp1fndgsc.Appx`. With this new failed test, the `.appx` file is now `Ubuntu_1804.2019.522.0_x64.appx`. So, it looks like MS updated the version of WSL they distribute, and that caused the Chocolatey package to fail.

More information about this error can be found here: https://www.gep13.co.uk/blog/chocolatey-error-hashes-do-not-match. It recommends: 

> The best way to fix this problem would be to reach out to the maintainers of the package and ask them to push a new package version that includes the correct checksum. In the case of this package, there is actually a new version of the application available, so this package is due to be updated. If there wasn't a new version available, then the maintainer could push a new package version using what is known as the package fix notation.

File: setup.py [hunk 1]
```diff
@@ -38,80 +38,85 @@
     keywords='Magnetic Resonance Imaging MRI spinal cord analysis template',
     packages=find_packages(exclude=['.git', 'data', 'dev', 'dev.*',
                                     'install', 'testing']),
+    data_files=[
+        # <hack>
+        ("sct_scripts", [os.path.join("scripts", x) for x in os.listdir("scripts") if x.endswith(".py")]),
+        # </hack>
+    ],
```
Comment (kousu) [inline setup.py:45]: Was this on purpose? It's in b10c820c809087794536e9fe88b61b6eb443522f which doesn't explain it in its commit message.

I would expect `console_scripts` handles making sure these get included in the right places.

Comment (joshuacwnewton) [reply setup.py:45]: This was recently removed, in #2837, but it looks like it snuck back in.

Comment (jcohenadad) [reply setup.py:45]: 🤭my bad

Comment (kousu) [reply setup.py:45]: That's what code review is for!

File: MANIFEST.in [hunk 2]
```diff
@@ -3,7 +3,7 @@ include setup*
 include README*
 include CHANGES.md
 include abstract.png
-include version.txt
+include spinalcordtoolbox/version.txt
```
Comment (kousu) [inline MANIFEST.in:6]: I did some testing and found that nothing in this file was being respected. I don't know if it ever was. According to my experiments any file outside of a package (as defined in setup.py:

>     packages=find_packages(exclude=['.git', 'data', 'dev', 'dev.*', 'install', 'testing']),

) is just ignored.

This patch here does do something but it's the only line that does.

An equivalent patch, I believe, would be

```
(.venv) [kousu@requiem spinalcordtoolbox]$ git diff
diff --git a/setup.py b/setup.py
index 8057fa56..0e808c41 100644
--- a/setup.py
+++ b/setup.py
@@ -39,6 +39,9 @@ setup(
     packages=find_packages(exclude=['.git', 'data', 'dev', 'dev.*',
                                     'install', 'testing']),
     include_package_data=True,
+    package_data={
+        "spinalcordtoolbox": ["version.txt"],
+    },
     extras_require={
         'docs': [
         'sphinx',
``` 

+

`git rm MANIFEST.in`

**Pull request merged**